hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableScan.java

```diff
@@ -0,0 +1,116 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive.legacy;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import java.util.Collection;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.iceberg.BaseFileScanTask;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.DataTableScan;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.Metrics;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.PartitionSpecParser;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ResidualEvaluator;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.util.ParallelIterable;
+import org.apache.iceberg.util.ThreadPools;
+
+
+/**
+ * A {@link DataTableScan} which uses Hive table and partition metadata to read tables.
+ * This scan does not provide any time travel, snapshot isolation, incremental computation benefits.
+ */
+public class LegacyHiveTableScan extends DataTableScan {
+
+  protected LegacyHiveTableScan(TableOperations ops, Table table) {
+    super(ops, table);
+  }
+
+  protected LegacyHiveTableScan(TableOperations ops, Table table, Long snapshotId, Schema schema, Expression rowFilter,
+      boolean caseSensitive, boolean colStats, Collection<String> selectedColumns,
+      ImmutableMap<String, String> options) {
+    super(ops, table, snapshotId, schema, rowFilter, caseSensitive, colStats, selectedColumns, options);
+  }
+
+  @Override
+  @SuppressWarnings("checkstyle:HiddenField")
+  protected TableScan newRefinedScan(TableOperations ops, Table table, Long snapshotId, Schema schema,
+      Expression rowFilter, boolean caseSensitive, boolean colStats, Collection<String> selectedColumns,
+      ImmutableMap<String, String> options) {
+    return new LegacyHiveTableScan(ops, table, snapshotId, schema, rowFilter, caseSensitive, colStats, selectedColumns,
+        options);
+  }
+
+  @Override
+  public CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot, Expression rowFilter,
+      boolean caseSensitive, boolean colStats) {
+    PartitionSpec spec = ops.current().spec();
+    String schemaString = SchemaParser.toJson(spec.schema());
+    String specString = PartitionSpecParser.toJson(spec);
+    // TODO: Consider returning the whole rowFilter as residual since we many not be able to guarantee that all
```

Comment:
+1

Reply:
After revisiting this I realized that the residuals are only used for PPD supported by the file formats. Iceberg returns all filters to be reapplied by the engine anyway. So this should not be issue. https://github.com/linkedin/iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/Reader.java#L245

Reply:
Removed the TODO